src/main/java/net/sf/jsqlparser/parser/ParserKeywordsUtils.java

```diff
@@ -0,0 +1,429 @@
+
+                // add keywords from the composite token definitions:
+                // tk=<K_DATE_LITERAL> | tk=<K_DATETIMELITERAL> | tk=<K_STRING_FUNCTION_NAME>
+                // we will use the composite tokens instead, which are always hit first before the simple keywords
+                // @todo: figure out a way to remove these composite tokens, as they do more harm than good
+                , { "SEL", RESTRICTED_JSQLPARSER }
+                , { "SELECT", RESTRICTED_JSQLPARSER }
+
+                , { "DATE", RESTRICTED_JSQLPARSER }
+                , { "TIME" , RESTRICTED_JSQLPARSER }
+                , { "TIMESTAMP", RESTRICTED_JSQLPARSER }
+
+                , { "YEAR", RESTRICTED_JSQLPARSER }
+                , { "MONTH", RESTRICTED_JSQLPARSER }
+                , { "DAY", RESTRICTED_JSQLPARSER }
+                , { "HOUR", RESTRICTED_JSQLPARSER }
+                , { "MINUTE" , RESTRICTED_JSQLPARSER }
+                , { "SECOND", RESTRICTED_JSQLPARSER }
+
+                , { "SUBSTR", RESTRICTED_JSQLPARSER }
+                , { "SUBSTRING", RESTRICTED_JSQLPARSER }
+                , { "TRIM", RESTRICTED_JSQLPARSER }
+                , { "POSITION", RESTRICTED_JSQLPARSER }
+                , { "OVERLAY", RESTRICTED_JSQLPARSER }
+
+                , { "NEXTVAL", RESTRICTED_JSQLPARSER }
+
+                //@todo: figure out what those are about
+                , { "RR", RESTRICTED_JSQLPARSER }
+                , { "RS", RESTRICTED_JSQLPARSER }
+                , { "UR", RESTRICTED_JSQLPARSER }
+                , { "CS", RESTRICTED_JSQLPARSER }
+
+                //@todo: Object Names should not start with Hex-Prefix, we shall not find that Token
+                , { "0x", RESTRICTED_JSQLPARSER }
+        };
+
+        ArrayList<String> keywords = new ArrayList<>();
+        for (Object[] data : ALL_RESERVED_KEYWORDS) {
+            int value = (int) data[1];
+
+            // test if bit is not set
+            if ( (value & restriction) == restriction
+                    || (restriction & value) == value ) {
+                keywords.add((String) data[0]);
+            }
+        }
+
+        return keywords;
+    }
+
+    public static void main(String[] args) throws Exception {
+        if (args.length<1) {
+            throw new IllegalArgumentException("No filename provided as parameters ARGS[0]");
+        }
+
+        File file = new File(args[0]);
+        if (file.exists() && file.canRead()) {
+            buildGrammarForRelObjectName(file);
+            buildGrammarForRelObjectNameWithoutValue(file);
+        } else {
+            throw new FileNotFoundException("Can't read file " + args[0]);
+        }
+    }
+
+    public static TreeSet<String> getAllKeywordsUsingRegex(File file) throws IOException {
+        Pattern tokenBlockPattern = Pattern.compile("TOKEN\\s*:\\s*(?:/\\*.*\\*/*)\\n\\{(?:[^\\}\\{]+|\\{(?:[^\\}\\{]+|\\{[^\\}\\{]*\\})*\\})*\\}", Pattern.MULTILINE);
```

Comment:
Cool regex. Maybe insert a comment to indicate/example what matches? I cannot read it from the regex directly.